### PR TITLE
feat(value-list-item, pick-list-item): add disableSelect prop

### PR DESCRIPTION
**Pull request by asangma**
**Related Issue:** #2040

## Summary
Adds a disableSelect prop that disables selection of the item

The issue only calls for an enhancement to ValueListItem but in order to add the feature, it appeared that the update needed to also be done on PickListItem.

### CODE REVIEW

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx [hunk 1]
```diff
@@ -49,6 +49,11 @@ export class CalcitePickListItem {
    * When false, the item cannot be deselected by user interaction.
    */
   @Prop() disableDeselect = false;
+  
+  /**
+   * When true, the item cannot be selected by user interaction.
+   */
+  @Prop({ reflect: true }) disableSelect = false;
```
Comment (driskull) [inline src/components/calcite-pick-list-item/calcite-pick-list-item.tsx:56]: The name of this prop is a bit confusing. Right above it we have `disableDeselect` and this is `disableSelect`. It sounds like they would conflict? Any way these can be merged or named differently?

I thought for naming we had proposed names like `selectDisabled`?

**Review by driskull — COMMENTED** [review]
Comments

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx (hunk 1 above)
Comment (asangma) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx:56]: Actually, thinking about it now, we should probably use what we used in ListItem: `nonInteractive`. Yeah?

Comment (driskull) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx:56]: yeah

Comment (asangma) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: @driskull Updated.

**Review by driskull — APPROVED** [review]
This looks good to me. 🌈

**Comment by asangma** [comment]
Thanks @driskull.

**Pull request merged**